buffer/src/test/java/io/netty/buffer/AbstractCompositeByteBufTest.java

```diff
@@ -1443,4 +1443,70 @@ private static void assertCompositeBufCreated(int expectedMaxComponents) {
         assertTrue(buf.release());
     }
 
+    @Test
+    public void testDiscardSomeReadBytesCorrectlyUpdatesLastAccessed() {
+        testDiscardCorrectlyUpdatesLastAccessed(true);
+    }
+
+    @Test
+    public void testDiscardReadBytesCorrectlyUpdatesLastAccessed() {
+        testDiscardCorrectlyUpdatesLastAccessed(false);
+    }
+
+    private static void testDiscardCorrectlyUpdatesLastAccessed(boolean discardSome) {
+        CompositeByteBuf cbuf = compositeBuffer();
+        List<ByteBuf> buffers = new ArrayList<ByteBuf>(4);
+        for (int i = 0; i < 4; i++) {
+            ByteBuf buf = buffer().writeInt(i);
+            cbuf.capacity(cbuf.writerIndex()).addComponent(true, buf);
```

Comment:
Curious why `.capacity(cbuf.writerIndex())` is needed here?

Reply:
actually it is not... copy and paste. Removed